app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -1,25 +1,17 @@
 module MfaSetupConcern
   extend ActiveSupport::Concern
 
-  def user_next_authentication_setup_path(next_setup_choice)
-    if user_session.dig(:selected_mfa_options, determine_next_mfa_selection).present? &&
-       IdentityConfig.store.select_multiple_mfa_options
+  def next_setup_path
+    if user_needs_confirmation_screen?
       auth_method_confirmation_url(next_setup_choice: next_setup_choice)
     else
-      user_session.delete(:selected_mfa_options)
+      user_session.delete(:mfa_selections)
       nil
     end
   end
 
-  def determine_next_mfa_selection
-    return unless user_session[:selected_mfa_options]
-    current_session = user_session[:next_mfa_selection_choice]
-    current_index = user_session[:selected_mfa_options].find_index(current_session) || 0
-    current_index + 1
-  end
-
   def confirmation_path(next_mfa_selection_choice)
-    user_session[:next_mfa_selection_choice] = next_mfa_selection_choice
+    user_session[:next_mfa_setup] = next_mfa_selection_choice
```

Comment:
in general, renaming keys in session data like this is a little dicey, because our deploys take ~15 minutes and a user could be hitting either a new or an old server and that server might not have the data in the session it expects.

Is it possible to avoid renaming it here?